netbox/netbox/api/viewsets/__init__.py

```diff
@@ -157,3 +159,22 @@ def perform_destroy(self, instance):
         logger.info(f"Deleting {model._meta.verbose_name} {instance} (PK: {instance.pk})")
 
         return super().perform_destroy(instance)
+
+
+class MPTTLockedMixin(GenericViewSet):
```

Comment:
Does this need to inherit from `GenericViewSet`? Seems like it would be fine as a base class.